### PR TITLE
mounter(ticdc): remove mounter worker

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref #5051

### What is changed and how it works?
this pr remove mounter worker and decode event in sortNode directly,  

 it increased the throughput of the table pipeline, the see test result below
1 table and 1 cdc server , and changefeed with blackhole sink
with mounter worker,  the throughput is 64k
without mounter worker, the throughput is more than 84k

5bzlNabjah

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/entry/mounter.go [hunk 1]
```diff
@@ -126,60 +108,35 @@ func (m *mounterImpl) Run(ctx context.Context) error {
...
 		}
 	}
 }
 
-func (m *mounterImpl) codecWorker(ctx context.Context, index int) error {
-	changefeedID := util.ChangefeedIDFromCtx(ctx)
-	metricMountDuration := mountDuration.WithLabelValues(changefeedID)
-	metricTotalRows := totalRowsCountGauge.WithLabelValues(changefeedID)
+func (m *mounterImpl) DecodeEvent(ctx context.Context, pEvent *model.PolymorphicEvent) error {
+	m.metricMounterInputChanSize.Inc()
```
Comment (overvenus) [inline cdc/entry/mounter.go:119]: Let's remove it if there is no channel at all.

File: cdc/model/mounter.go [hunk 2]
```diff
@@ -13,19 +13,14 @@
...
-
 // PolymorphicEvent describes an event can be in multiple states
 type PolymorphicEvent struct {
 	StartTs uint64
 	// Commit or resolved TS
 	CRTs uint64
 
-	RawKV    *RawKVEntry
-	Row      *RowChangedEvent
-	finished chan struct{}
```
Comment (overvenus) [inline cdc/model/mounter.go:28]: Great! We finally removed it.

File: cdc/entry/mounter.go [hunk 3]
```diff
@@ -69,55 +62,44 @@ type rowKVEntry struct {
 // Mounter is used to parse SQL events from KV events
 type Mounter interface {
 	Run(ctx context.Context) error
```
Comment (overvenus) [inline cdc/entry/mounter.go:64]: Can we remove `Run` in the interface?

**Comment by overvenus** [comment]
Please also update PR description and labels, thanks!

**Comment by ben1009** [comment]
could you provide more context why make the change, thx

File: cdc/entry/mounter.go (hunk 3 above)
Comment (sdojjy) [reply cdc/entry/mounter.go:64]: done

File: cdc/entry/mounter.go (hunk 1 above)
Comment (sdojjy) [reply cdc/entry/mounter.go:119]: done

File: cdc/entry/mounter.go [hunk 4]
```diff
@@ -68,56 +61,51 @@ type rowKVEntry struct {
...
+		enableOldValue:      enableOldValue,
+		metricMountDuration: mountDuration.WithLabelValues(changefeedID),
+		metricTotalRows:     totalRowsCountGauge.WithLabelValues(changefeedID),
+		tz:                  tz,
 	}
 }
 
-const defaultMounterWorkerNum = 32
-
 func (m *mounterImpl) Run(ctx context.Context) error {
```
Comment (liuzix) [inline cdc/entry/mounter.go:99]: Do we still need this method?

File: cdc/entry/mounter.go [hunk 5]
```diff
@@ -126,60 +114,31 @@ func (m *mounterImpl) Run(ctx context.Context) error {
...
+	start := time.Now()
+	rowEvent, err := m.unmarshalAndMountRowChanged(ctx, pEvent.RawKV)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	pEvent.Row = rowEvent
+	pEvent.RawKV.Value = nil
+	pEvent.RawKV.OldValue = nil
+	duration := time.Since(start)
+	if duration > time.Second {
```
Comment (liuzix) [inline cdc/entry/mounter.go:138]: Could this condition produce misleading metrics results?

File: cdc/entry/mounter.go [hunk 6]
```diff
@@ -126,60 +114,31 @@ func (m *mounterImpl) Run(ctx context.Context) error {
...
-	}
-}
-
-func (m *mounterImpl) AddEntry(ctx context.Context, event *model.PolymorphicEvent) error {
-	index := atomic.AddInt64(&m.index, 1) % int64(m.workerNum)
-	select {
-	case <-ctx.Done():
-		return ctx.Err()
-	case m.rawRowChangedChs[index] <- event:
+func (m *mounterImpl) DecodeEvent(ctx context.Context, pEvent *model.PolymorphicEvent) error {
```
Comment (liuzix) [inline cdc/entry/mounter.go:124]: Please add a note stating that this method could block indefinitely if the DDL puller is lagging.

File: cdc/entry/mounter.go (hunk 4 above)
Comment (sdojjy) [reply cdc/entry/mounter.go:99]: no, removed

File: cdc/entry/mounter.go (hunk 5 above)
Comment (sdojjy) [reply cdc/entry/mounter.go:138]: yes, but according to this PR: https://github.com/pingcap/tiflow/pull/4585,  it's better to reduce some metrics overhead.

File: cdc/entry/mounter.go (hunk 6 above)
Comment (sdojjy) [reply cdc/entry/mounter.go:100]: done

File: cdc/entry/metrics.go [hunk 7]
```diff
@@ -18,13 +18,6 @@ import (
 )
 
 var (
-	mounterInputChanSizeGauge = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Namespace: "ticdc",
-			Subsystem: "mounter",
-			Name:      "input_chan_size",
```
Comment (overvenus) [inline cdc/entry/metrics.go:25]: Could you also update ticdc.json to remove it from grafana?

Comment (overvenus) [reply cdc/entry/metrics.go:25]: Also the data-flow.dot file in /docs.

**Review by overvenus — APPROVED** [review]
rest LGTM

File: cdc/entry/metrics.go (hunk 7 above)
Comment (sdojjy) [reply cdc/entry/metrics.go:25]: done

**Review by liuzix — APPROVED** [review]

**Pull request merged**